### PR TITLE
docs: Update agent tool documentation and enhance flow as tool component

### PR DESCRIPTION
**Pull request by mendonk**
Documentation improvements:

* [`docs/docs/Agents/agent-tool-calling-agent-component.md`](diffhunk://#diff-20ba5ad0663ae0def401ee3cacb7193272bf5a8956857c12cb377b64e8a711deL151-L154): Removed redundant instructions related to the **Tool Mode** button to streamline the documentation.
* [`docs/docs/Agents/agent-tool-calling-agent-component.md`](diffhunk://#diff-20ba5ad0663ae0def401ee3cacb7193272bf5a8956857c12cb377b64e8a711deL184-R180): Clarified the usage of flows as tools by linking to the `flow as tool` component documentation.

### CODE REVIEW

File: docs/docs/Components/components-logic.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+---
+title: Logic
```
Comment (KimberlyFields) [inline docs/docs/Components/components-logic.md:2]: What about logic? Update the title to make it more descriptive.

File: docs/docs/Agents/agent-tool-calling-agent-component.md [hunk 2]
```diff
@@ -181,7 +177,7 @@ inputs = [
 
 ## Add flows as tools
 
-An agent can use flows that are saved in your workspace as tools.
+An agent can use flows that are saved in your workspace as tools with the [flow as tool](/components-logic#flow-as-tool) component.
```
Comment (KimberlyFields) [inline docs/docs/Agents/agent-tool-calling-agent-component.md:180]: How is this capitalized in the UI? You don't have it capitalized here, but you have initial caps on it in the next file.

**Review by KimberlyFields — APPROVED** [review]
Left a few comments, but approving.

File: docs/docs/Components/components-logic.md (hunk 1 above)
Comment (mendonk) [reply docs/docs/Components/components-logic.md:2]: Title is below

**Pull request merged**